automatic/sysinternals/tools/chocolateyInstall.ps1

```diff
@@ -21,6 +21,23 @@ if (Is-NanoServer) {
   $packageArgs.checksum = '7b6fab7f0ada0bc2b0748f3bc29e93846869f9dd175a0ca070bbebc07ee5de09'
  }
 
+if (-Not $pp.NoStartMenu) {
+  Write-Verbose "Creating start menu entry"
+  set-alias 7z $Env:chocolateyInstall\tools\7z.exe
+  $linkfiles = get-childitem "$installDir" -include "*.exe" -recurse
+  $startmenufolder = "$($env:ProgramData)\Microsoft\Windows\Start Menu\Programs\Sysinternals Suite"
+  Add-Content "$toolsPath\chocoUninstall.ps1" -Value "Remove-Item `"$($startmenufolder)`" -Recurse -Force -ea ignore"
+  # This will create a Link for all gui application and add a line chocoUninstall.ps1 to remove it on uninstall
+  foreach ($file in $linkfiles) {
+    $guiinf = 7z l $file
```

Comment:
Using `7z` directly is not acceptable.
If needed, you should rather iterate over the already extracted files.

Reply:
This detects where it is a GUI or a console application. Adding a start menu entry fro a console app seems unreasonable. Besides hacking directly in the PE, using 7z seems to be the cleanest solution. Parsing the output of shimgen seems worse.....

Reply:
I don't like this - Calling external app in iteration (7z) for bunch of other apps will slow down installation a lot. 

IMO you should use hardcoded list of GUI apps. The small inconvenience is that new additions would not be visible unless list is changed, but that is really tiny problem given the frequency new GUI tools get published.

Reply:
Just my two cents but this method of using 7zip creates dependency on 7zip

FYI:
The use of 7zip in the chocolatey tools is currently temporary and should not be used in any package. This usage by chocolatey could change in the future with our without notice.

Reply:
> Adding a start menu entry fro a console app seems unreasonable.

I agree with you here, there shouldn't be a need to do this for CLI apps.

> Besides hacking directly in the PE, using 7z seems to be the cleanest solution. Parsing the output of shimgen seems worse.....

I never said you needed to hack directly into the PE, or parsing the output of shimgen.
The files have already been extracted to disk at this point, so iterating over the already extracted files would be better (using `Get-ChilItem` for instance).
You are already just making a decision based on the name of the file, so to the end, it wouldn't be any different than what you already are doing (from a user perspective), other than you don't call 7z directly and don't open/close/reopen the archive several times.